File: client/makc2021-react-redux-ts/src/Layer5/Pages/DummyMain/List/Queries/Get/DummyMainItemPageGetQueryInput.ts
// Copyright (c) 2021 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

import {
  createListGetQueryDomainInput,
  ListGetQueryDomainInput
} from 'src/Layer4/Domains/DummyMain/Queries/List/Get/ListGetQueryDomainInput';

/**
 * Входные данные запроса на получение страницы сущностей "DummyMain".
 */
export interface DummyMainListPageGetQueryInput {
  /**
   * Список.
   */
  list: ListGetQueryDomainInput;
}

export function createDummyMainListPageGetQueryInput(): DummyMainListPageGetQueryInput {
  return {
    list: createListGetQueryDomainInput()
  } as DummyMainListPageGetQueryInput;
}

File: server/src/Makc2021.Layer5.Apps.API/Pages/DummyMain/Item/Queries/Get/DummyMainItemPageGetQueryInput.cs
// Copyright (c) 2021 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using Makc2021.Layer4.Domains.DummyMain.Queries.Item.Get;

namespace Makc2021.Layer5.Apps.API.Pages.DummyMain.Item.Queries.Get
{
    /// <summary>
    /// Входные данные запроса на получение страницы сущности "DummyMain".
    /// </summary>
    public class DummyMainItemPageGetQueryInput
    {
        #region Properties

        /// <summary>
        /// Элемент.
        /// </summary>
        public ItemGetQueryDomainInput Item { get; } = new();

        #endregion Properties
    }
}
